control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1515,12 +1555,16 @@ P4RuntimeSerializer::serializeP4RuntimeIfRequired(const IR::P4Program* program,
 
     auto p4Runtime = get()->generateP4Runtime(program, arch);
 
-    std::ostream* out = openFile(options.p4RuntimeFile, false);
-    if (!out) {
-        ::error("Couldn't open P4Runtime API file: %1%", options.p4RuntimeFile);
-        return;
+    for (unsigned i = 0; i < files.size(); i++) {
+        cstring file = files.at(i);
+        P4::P4RuntimeFormat format = formats.at(i);
+        std::ostream* out = openFile(file, false);
+        if (!out) {
+            ::error("Couldn't open P4Runtime API file: %1%", file);
+            continue;
+        }
+        p4Runtime.serializeP4InfoTo(out, format);
     }
-    p4Runtime.serializeP4InfoTo(out, options.p4RuntimeFormat);
 
     if (!options.p4RuntimeEntriesFile.isNullOrEmpty()) {
```

Comment:
we use `options.p4RuntimeFormat` here:
```
p4Runtime.serializeEntriesTo(out, options.p4RuntimeFormat);
```
It seems that maybe the same should be done for the entries file: add `--p4runtime-entries-files` and detect the format based one the extension?